test/controllers/authentications/microsoft_graph_oauth_controller_test.rb

```diff
@@ -0,0 +1,182 @@
+require "test_helper"
+
+class Authentications::MicrosoftGraphOauthControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    OmniAuth.config.test_mode = true
+  end
+
+  test "cancelling a microsoft oauth flow redirects to the settings page" do
+    login_as users(:keith)
+    get "/auth/failure"
+    assert_redirected_to edit_settings_person_path
+  end
+
+  test "cancelling a microsoft oauth flow redirects to the login" do
+    get "/auth/failure"
+    assert_redirected_to login_path
+  end
```

Comment:
Does Microsoft ever redirect here? If you look at the first block `omniauth.rb`, that's where google auth configures the failure path. I don't see an equivalent configuration of failure path for Microsoft.

If you start logging in or signing up with Microsoft oauth and then you abort on the Microsoft confirmation step, what happens? I don't think it'll redirect here unless we tell it, but I'm curious where it goes by default.

Reply:
Good question.

Cancelling login/auth looks to redirect to: `/auth/microsoft_graph/callback?error=access_denied&error_description=The%20user%20has%20denied%20access%20to%20the%20scope%20requested%20by%20the%20client%20application.&state=<state>`.

Adding/fixing tests.

Reply:
Done